apps/astarte_realm_management/test/astarte_realm_management/engine_test.exs

```diff
@@ -833,6 +854,16 @@ defmodule Astarte.RealmManagement.EngineTest do
     assert Engine.delete_interface("autotestrealm", "com.ObjectAggregation", 0) == :ok
   end
 
+  test "fail to install interface when database_retention_ttl is higher than the maximum" do
```

Comment:
Also add a success test (i.e. a realm with an explicit `datastream_maximum_storage_retention` + an interface with a lower TTL) to test all the combinations